src/book.js

```diff
@@ -653,78 +677,104 @@ export class Book {
+     */
+    calculatelayout(alt_folio){
+        // vampire
+        let l = this.calculate_dimensions()
+        let cellWidth = l.layoutCell[0]
+        let cellHeight = l.layoutCell[1]
+        let positions = []
 
-                // amount to inset by, relative to fore edge, on left side of book
-                let xForeEdgeShift = padding['fore_edge'] + ((this.page_positioning == 'centered' ) ? leftRightPageGap : 2 * leftRightPageGap);
-                // amount to inset by, relative to binding, on right side of book
-                let xBindingShift = padding['binding'] + ((this.page_positioning == 'centered' ) ? leftRightPageGap : 0);
+        l.layout.rotations.forEach((row, i) => {
+            row.forEach((col, j) => {
+                let xForeEdgeShift = l.xForeEdgeShiftFunc();
+                let xBindingShift = l.xBindingShiftFunc();
+                let yTopShift = l.yTopShiftFunc();
+                let yBottomShift = l.yBottomShiftFunc();
 
                 let isLeftPage = j % 2 == 0; //page on 'left' side of open book
-                let x = (j * finalx) + ((j % 2 == 0 ) ? xForeEdgeShift : xBindingShift);
-                let y = (i * finaly) + ypad + padding['bottom'] * sy;
+                let x = (j * cellWidth) + ((isLeftPage) ? xForeEdgeShift : xBindingShift);
+                let y = (i * cellHeight) + yBottomShift;
 
                 if ([-180].includes(col)) { // upside-down page
-                    let isLeftPage = j % 2 == 1; //page on 'left' (right side on screen)
-                    y = finaly + (i * finaly) - ypad - padding['bottom'] * sy;
-                    x = finalx + (j * finalx) - ((j % 2 == 0) ? xBindingShift : xForeEdgeShift);
-                }
-
-                if ([90].includes(col)) {   // 'top' of page is on left, right side of screen
-                    let isLeftPage = i % 2 == 0; // page is on 'left' (top side of screen)
-                    x = (1 + j) * finalx - padding['bottom'] - xpad;// + padding['top'];
-                    y = (i * finaly) + ((isLeftPage) ? xForeEdgeShift : xBindingShift);// + ((isLeftPage) ? 2 * ypad : 0);//(this.padding_pt['binding'] * sx * 2));
-                }
-                if ([-90].includes(col)) {  // 'top' of page is on the right, left sight of screen
-                    let isLeftPage = i % 2 == 1; // page is on 'left' (bottom side of screen)
-                    x = j * finalx + padding['bottom'] + xpad;
-                    y = ((1+i) * finaly) - ((isLeftPage) ? xForeEdgeShift : xBindingShift);// - ((isLeftPage) ? 0 + this.padding_pt['binding'] * sx : (2 * ypad) - (this.padding_pt['binding'] * sx));
+                    isLeftPage = j % 2 == 1; //page on 'left' (right side on screen)
+                    y = (i + 1) * cellHeight - yBottomShift;
+                    x = (j + 1) * cellWidth - ((isLeftPage) ? xForeEdgeShift : xBindingShift);
+
+                } else if ([90].includes(col)) {   // 'top' of page is on left, right side of screen
```

Comment:
Same comment here and for 767 as on 757 about using equality rather than `includes`